python/pyspark/sql/connect/column.py

```diff
@@ -73,20 +114,6 @@ def __str__(self) -> str:
         return f"Literal({self._value})"
 
 
-def _bin_op(
-    name: str, doc: str = "binary function", reverse: bool = False
-) -> Callable[["ColumnRef", Any], Expression]:
-    def _(self: "ColumnRef", other: Any) -> Expression:
-        if isinstance(other, get_args(PrimitiveType)):
-            other = LiteralExpression(other)
-        if not reverse:
-            return ScalarFunctionExpression(name, self, other)
-        else:
-            return ScalarFunctionExpression(name, other, self)
-
-    return _
-
-
 class ColumnRef(Expression):
```

Comment:
I think we should rename this to `AttributeReference` to avoid confusion e.g., I think it's a bit mixed with `Column` interface that is the user-facing interface. @amaliujia and @cloud-fan 

Should better to keep it matched with either Catalyst internal types or user-facing Spark SQL interface classes.

Reply:
Let's have a discussion about this, but this is an unrelated change to this one. I think we should probably call `Expression` -> `Column` and `ColumnRef` -> `AttributeReference` but it will require some more digging what the right name should be. However, as said, that's independent of this change.

Reply:
+1 I have been thinking this `ColumnRef` thing. Let's revisit it on the naming, etc.